templates/boltzmann/core/honeycomb.ts

```diff
@@ -0,0 +1,909 @@
+     * ▔▏┗┛▕▔ we
+     * ╱▔▔▔▔▔▔▔▔▔▔╲
+     * trace with opentelemetry
+     * ╱╱┏┳┓╭╮┏┳┓ ╲╲
+     * ▔▏┗┻┛┃┃┗┻┛▕▔
+     */
+
+    const tracer = this.tracer
+    let carrier = {}
+
+    // Start a parent span
+    const parentSpan = tracer.startSpan(
+      `${context.method} ${context.url.pathname}${context.url.search}`,
+      {
+        attributes: {
+          [SemanticAttributes.HTTP_HOST]: context.host,
+          [SemanticAttributes.HTTP_URL]: context.url.href,
+          [SemanticAttributes.NET_PEER_IP]: context.remote,
+          [SemanticAttributes.HTTP_METHOD]: context.method,
+          [SemanticAttributes.HTTP_SCHEME]: context.url.protocol,
+          [SemanticAttributes.HTTP_ROUTE]: context.url.pathname,
+          // TODO: Honeycomb.Attributes obvs
+          [Honeycomb.OTEL_REQ_QUERY]: context.url.search
+        }
+      }
+    )
+
+    // this propagator takes care of extracting trace parent
+    // and state from request headers (and so on)
+    const propagator = new W3CTraceContextPropagator()
+
+    propagator.inject(
+      trace.setSpanContext(
+        ROOT_CONTEXT,
+        parentSpan.spanContext()
+      ),
+      carrier,
+      defaultTextMapSetter
+    )
+
+    /* TODO: Do I need to create and set a context? No, right?
+
+    // create a parent active context
+    const parentContext = propagator.extract(
+      ROOT_CONTEXT,
+      carrier,
+      defaultTextMapGetter
+    )
+
+    // set the active context
+    await traceContext.with(parentContext, async () => {
+
+    */
+
+    const rv = await runInContext()
+
+    const handler: Handler = <Handler>context.handler
+
+    parentSpan.setAttribute(
+      SemanticAttributes.HTTP_STATUS_CODE,
+      String(context._response.statusCode)
+    )
+    parentSpan.setAttribute(
+      SemanticAttributes.HTTP_ROUTE,
+      <string>handler.route
+    )
+    parentSpan.setAttribute(
+      SemanticAttributes.HTTP_METHOD,
+      <string>handler.method
+    )
+    parentSpan.setAttribute(
+      SemanticResourceAttributes.SERVICE_VERSION,
+      <string>handler.version
+    )
+
+    Object.entries(context.params).map(([key, value]) => {
+      parentSpan.setAttribute(
+        Honeycomb.paramAttribute(key),
+        value
+      )
+    })
+    parentSpan.end()
+
+    return rv
+  }
+
+  private defaultHeaderSources: string[] = [ 'x-honeycomb-trace', 'x-request-id' ]
+
+  /* A beelines implementation of startTrace.
+   */
+  async withBeelineTrace(
+    context: Context,
+    runInContext: () => Promise<void>,
+    headerSources?: string[]
+  ): Promise<any> {
+    const schema = require('honeycomb-beeline/lib/schema')
+    const tracker = require('honeycomb-beeline/lib/async_tracker')
+
+    const traceContext = _getBeelineTraceContext(context)
+    const trace = beeline.startTrace({
+      [schema.EVENT_TYPE]: 'boltzmann',
+      [schema.PACKAGE_VERSION]: '1.0.0',
+      [schema.TRACE_SPAN_NAME]: `${context.method} ${context.url.pathname}${context.url.search}`,
```

Comment:
I was talking with ceej, and we're pretty sure that we want to remove `${context.url.search}` here (and also in the same line of the beeline code)
we don't want the query params as part of the span name, that will cause issues.

We should probably take a look at the span names that are being created once you have the test stuff up and running, and then take care of this.